src/NosCore.Parser/Parsers/QuestParser.cs

```diff
@@ -0,0 +1,293 @@
+    //DATA	0	1	-1	1
+    //PRIZE	-1	-1	-1	-1
+    //LINK	2
+    //END
+    //
+    //#=======
+
+    public class QuestParser
+    {
+        private readonly string _fileQuestDat = "\\quest.dat";
+        private readonly ILogger _logger;
+        private readonly IGenericDao<QuestDto> _questDao;
+        private readonly IGenericDao<QuestObjectiveDto> _questObjectiveDao;
+
+
+        public QuestParser(IGenericDao<QuestDto> questDao, IGenericDao<QuestObjectiveDto> questObjectiveDao, ILogger logger)
+        {
+            _logger = logger;
+            _questDao = questDao;
+            _questObjectiveDao = questObjectiveDao;
+        }
+
+        public void ImportQuests(string folder)
+        {
+            var actionList = new Dictionary<string, Func<Dictionary<string, string[][]>, object>>
+            {
+                {nameof(QuestDto.QuestId), chunk => Convert.ToInt16(chunk["VNUM"][0][1])},
+                {nameof(QuestDto.QuestType), chunk => Convert.ToInt32(chunk["VNUM"][0][2])},
+                {nameof(QuestDto.LevelMin), chunk => Convert.ToByte(chunk["LEVEL"][0][1])},
+                {nameof(QuestDto.LevelMax), chunk => Convert.ToByte(chunk["LEVEL"][0][2])},
+                {nameof(QuestDto.Title), chunk => chunk["TITLE"][0][1]},
+                {nameof(QuestDto.Desc), chunk => chunk["DESC"][0][1]},
+                {nameof(QuestDto.TargetX), chunk =>  chunk["TARGET"][0][1] == "-1" ? (short?)null : Convert.ToInt16(chunk["TARGET"][0][1])},
+                {nameof(QuestDto.TargetY), chunk =>  chunk["TARGET"][0][2] == "-1"  ? (short?)null : Convert.ToInt16(chunk["TARGET"][0][2])},
+                {nameof(QuestDto.TargetMap), chunk => chunk["TARGET"][0][3] == "-1"  ? (short?)null : Convert.ToInt16(chunk["TARGET"][0][3])},
+                {nameof(QuestDto.StartDialogId), chunk => chunk["TARGET"][0][1] == "-1" ? (int?)null :  Convert.ToInt32(chunk["TALK"][0][1])},
+                {nameof(QuestDto.EndDialogId), chunk => chunk["TARGET"][0][2] == "-1" ? (int?)null :  Convert.ToInt32(chunk["TALK"][0][2])},
+            };
+            var genericParser = new GenericParser<QuestDto>(folder + _fileQuestDat, "#=======", 0, actionList, _logger);
+            var quests = genericParser.GetDtos("    ");
+
+            _questDao.InsertOrUpdate(quests);
+            _questObjectiveDao.InsertOrUpdate(quests.Where(s => s.QuestObjective != null).SelectMany(s => s.QuestObjective));
+
+            _logger.Information(LogLanguage.Instance.GetMessageFromKey(LogLanguageKey.QUESTS_PARSED), quests.Count);
+        }
+    }
+
+    //public void ImportQuests()
+    //{
+    //                switch (currentLine[0])
+    //                {
+    //                    case "VNUM":
+    //                        switch (quest.QuestId)
+    //                        {
+    //                            //TODO: Legendary Hunter quests will be markes ad daily, but should be in the "secondary" slot and be daily nevertheless
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/app/NosCoreIO/NosCore/pullRequest?prid=4634362)